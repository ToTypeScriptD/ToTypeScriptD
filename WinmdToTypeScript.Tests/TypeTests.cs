using ApprovalTests;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WinmdToTypeScript.Tests
{

    public class TypeTests : TestBase
    {
        [Test]
        public void EnumType()
        {
            var result = GetNativeType("SampleEnum").ToTypeScript();
            Approvals.Verify(result);
        }

        [Test]
        public void EnumTypeWithExplicitValues()
        {
            var result = GetNativeType("SampleEnumNumbered").ToTypeScript();
            Approvals.Verify(result);
        }

        [Test]
        public void ClassWithEventHandler()
        {
            var result = GetNativeType("ClassWithEventHandler").ToTypeScript();
            Approvals.Verify(result);
        }

        [Test]
        public void FullSampleAssembly()
        {
            var result = WinmdToTypeScript.Render.FullAssembly(base.NativeComponentPath);
            Approvals.Verify(result);
        }

        [Test]
        public void WindowsStorageClass()
        {
            var result = GetWinNativeType("Windows.Storage.StorageFile").ToTypeScript();
            Approvals.Verify(result);
        }
        [Test]
        public void WindowsSystemUserProfileGlobalizationPreferencesClass()
        {
            var result = GetWinNativeType("Windows.System.UserProfile.GlobalizationPreferences").ToTypeScript();
            Approvals.Verify(result);
        }
        
        //[Test]
        //public void FullWindowsAssembly()
        //{
        //    var file = @"C:\Windows\System32\WinMetadata\Windows.Foundation.winmd";
        //    var result = WinmdToTypeScript.Render.FullAssembly(file);
        //    Approvals.Verify(result);
        //}
    }
}
