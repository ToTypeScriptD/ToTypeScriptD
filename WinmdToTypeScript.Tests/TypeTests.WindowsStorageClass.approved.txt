declare module Windows.Storage {

    interface IStorageFile extends  Windows.Storage.IStorageItem,  Windows.Storage.Streams.IRandomAccessStreamReference,  Windows.Storage.Streams.IInputStreamReference {
        contentType: string;
        fileType: string;
        openAsync(accessMode: FileAccessMode, ): WinJS.Promise<Windows.Storage.Streams.IRandomAccessStream>;
        openTransactedWriteAsync(): WinJS.Promise<Windows.Storage.StorageStreamTransaction>;
        copyAsync(destinationFolder: IStorageFolder, ): WinJS.Promise<Windows.Storage.StorageFile>;
        copyAsync(destinationFolder: IStorageFolderdesiredNewName: string, ): WinJS.Promise<Windows.Storage.StorageFile>;
        copyAsync(destinationFolder: IStorageFolderdesiredNewName: stringoption: NameCollisionOption, ): WinJS.Promise<Windows.Storage.StorageFile>;
        copyAndReplaceAsync(fileToReplace: IStorageFile, ): IAsyncAction;
        moveAsync(destinationFolder: IStorageFolder, ): IAsyncAction;
        moveAsync(destinationFolder: IStorageFolderdesiredNewName: string, ): IAsyncAction;
        moveAsync(destinationFolder: IStorageFolderdesiredNewName: stringoption: NameCollisionOption, ): IAsyncAction;
        moveAndReplaceAsync(fileToReplace: IStorageFile, ): IAsyncAction;
    }

    interface IStorageItem {
        attributes: FileAttributes;
        dateCreated: Date;
        name: string;
        path: string;
        renameAsync(desiredName: string, ): IAsyncAction;
        renameAsync(desiredName: stringoption: NameCollisionOption, ): IAsyncAction;
        deleteAsync(): IAsyncAction;
        deleteAsync(option: StorageDeleteOption, ): IAsyncAction;
        getBasicPropertiesAsync(): WinJS.Promise<Windows.Storage.FileProperties.BasicProperties>;
        isOfType(type: StorageItemTypes, ): Boolean;
    }

    interface IStorageItemProperties {
        displayName: string;
        displayType: string;
        folderRelativeId: string;
        properties: StorageItemContentProperties;
        getThumbnailAsync(mode: ThumbnailMode, ): WinJS.Promise<Windows.Storage.FileProperties.StorageItemThumbnail>;
        getThumbnailAsync(mode: ThumbnailModerequestedSize: UInt32, ): WinJS.Promise<Windows.Storage.FileProperties.StorageItemThumbnail>;
        getThumbnailAsync(mode: ThumbnailModerequestedSize: UInt32options: ThumbnailOptions, ): WinJS.Promise<Windows.Storage.FileProperties.StorageItemThumbnail>;
    }

    export class StorageFile extends IStorageFile, IStorageItem, IRandomAccessStreamReference, IInputStreamReference{
        displayName: string;
        displayType: string;
        folderRelativeId: string;
        properties: StorageItemContentProperties;
        openReadAsync(): IAsyncOperation`1;
        openSequentialReadAsync(): IAsyncOperation`1;
        getThumbnailAsync(mode: ThumbnailMode, ): IAsyncOperation`1;
        getThumbnailAsync(mode: ThumbnailModerequestedSize: UInt32, ): IAsyncOperation`1;
        getThumbnailAsync(mode: ThumbnailModerequestedSize: UInt32options: ThumbnailOptions, ): IAsyncOperation`1;
        getFileFromPathAsync(path: string, ): IAsyncOperation`1;
        getFileFromApplicationUriAsync(uri: Uri, ): IAsyncOperation`1;
        createStreamedFileAsync(displayNameWithExtension: stringdataRequested: StreamedFileDataRequestedHandlerthumbnail: IRandomAccessStreamReference, ): IAsyncOperation`1;
        replaceWithStreamedFileAsync(fileToReplace: IStorageFiledataRequested: StreamedFileDataRequestedHandlerthumbnail: IRandomAccessStreamReference, ): IAsyncOperation`1;
        createStreamedFileFromUriAsync(displayNameWithExtension: stringuri: Urithumbnail: IRandomAccessStreamReference, ): IAsyncOperation`1;
        replaceWithStreamedFileFromUriAsync(fileToReplace: IStorageFileuri: Urithumbnail: IRandomAccessStreamReference, ): IAsyncOperation`1;
    }

}
declare module Windows.Storage.Streams {

    interface IInputStreamReference {
        openSequentialReadAsync(): WinJS.Promise<Windows.Storage.Streams.IInputStream>;
    }

    interface IRandomAccessStreamReference {
        openReadAsync(): WinJS.Promise<Windows.Storage.Streams.IRandomAccessStreamWithContentType>;
    }

}
